vtm/src/org/oscim/layers/tile/vector/labeling/LabelTileLoaderHook.java

```diff
@@ -146,8 +153,11 @@ public boolean process(MapTile tile, RenderBuckets buckets, MapElement element,
                 float x = 0;
                 float y = 0;
                 if (centroid == null) {
-                    if (Parameters.POLY_LABEL) {
-                        centroid = PolyLabel.get(element);
+                    if(element.centerPosition != null) {
+                        centroid = element.centerPosition;
+                        // skip unnecessary calculations if centroid is outside of visible area
+                        if (centroid.x < 0 || centroid.x > Tile.SIZE || centroid.y < 0 || centroid.y > Tile.SIZE)
+                            return false;
```

Comment:
This can be moved on top of `if` statement below `labelPosition` check for optimal use.